src/lv_widgets/lv_dropdown.c

```diff
@@ -195,7 +200,113 @@ void lv_dropdown_set_options(lv_obj_t * ddlist, const char * options)
     ext->option_cnt++;   /*Last option has no `\n`*/
     ext->sel_opt_id      = 0;
     ext->sel_opt_id_orig = 0;
-    ext->options = options;
+    
+    /*Allocate space for the new text*/
+    size_t len = strlen(options) + 1;
+    if(ext->options != NULL && ext->static_txt == 0) {
+        lv_mem_free(ext->options);
+        ext->options = NULL;
+    }
+
+    ext->options = lv_mem_alloc(len);
+
+    LV_ASSERT_MEM(ext->options);
+    if(ext->options == NULL) return;
+
+    strcpy(ext->options, options);
+
+    /*Now the text is dynamically allocated*/
+    ext->static_txt = 0;
+}
+
+/**
+ * Set the options in a drop down list from a string
+ * @param ddlist pointer to drop down list object
+ * @param options a staic string with '\n' separated options. E.g. "One\nTwo\nThree"
+ */
+void lv_dropdown_set_static_options(lv_obj_t * ddlist, const char * options)
+{
+    LV_ASSERT_OBJ(ddlist, LV_OBJX_NAME);
+    LV_ASSERT_STR(options);
+
+    lv_dropdown_ext_t * ext = lv_obj_get_ext_attr(ddlist);
+
+    /*Count the '\n'-s to determine the number of options*/
+    ext->option_cnt = 0;
+    uint16_t i;
+    for(i = 0; options[i] != '\0'; i++) {
+        if(options[i] == '\n') ext->option_cnt++;
+    }
+    ext->option_cnt++;   /*Last option has no `\n`*/
+    ext->sel_opt_id      = 0;
+    ext->sel_opt_id_orig = 0;
+
+    if(ext->static_txt == 0 && ext->options != NULL) {
+        lv_mem_free(ext->options);
+        ext->options = NULL;
+    }
+
+    ext->static_txt = 1;
+    ext->options = (char *)options;
+}
+
+/**
+ * Add an options to a drop down list from a string.  Only works for dynamic options.
+ * @param ddlist pointer to drop down list object
+ * @param options a string without '\n'. E.g. "Four"
+ * @param position the insert position, indexed from 0, -1 = end of string
+ */
+void lv_dropdown_add_option(lv_obj_t * ddlist, const char * option, int pos)
+{
+    LV_ASSERT_OBJ(ddlist, LV_OBJX_NAME);
+    LV_ASSERT_STR(option);
+
+    lv_dropdown_ext_t * ext = lv_obj_get_ext_attr(ddlist);
+
+    /*Can not append to static options*/
+    if(ext->static_txt != 0) return;
+
+    /*Allocate space for the new option*/
+    size_t old_len = (ext->options == NULL) ? 0 : strlen(ext->options);
+	size_t ins_len = strlen(option);
+	size_t new_len = ins_len + old_len + 2; /* +2 for terminating NULL and possible \n */
+	lv_mem_monitor_t mem_mon;
+	lv_mem_monitor(&mem_mon);
```

Comment:
Why do you perform memory monitoring and defrag here?

Reply:
Sorry, debug code left in place.